package/MDAnalysis/analysis/gnm.py

```diff
@@ -154,6 +158,45 @@ def order_list(w):
     return list_map
 
 
+def neighbour_generator(positions, cutoff):
```

Comment:
Having good performance with cython for this isn't straight forward. The algorithm currently makes heavy use of python lists and their dynamic nature. If we want to improve performance significantly we need to switch to a C friendly data structure.